src/client/components/dashboard/SettingsModal.tsx

```diff
@@ -0,0 +1,412 @@
+      if (variables.collaboratorEmail === user?.email) {
+        queryClient.invalidateQueries('checkers')
+        onClose()
+        history.push('/dashboard')
+      } else refetchCollaborators()
+    },
+    onError: (err) => {
+      styledToast({
+        status: 'error',
+        description: getApiErrorMessage(err),
+      })
+    },
+  })
+
+  // Handle is active switch toggle
+  const onSwitchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setIsDisabled(true)
+    setActive.mutateAsync({
+      ...propChecker,
+      isActive: event.target.checked,
+    })
+  }
+
+  // Handle collaborator email input field on change
+  const handleInputChange = (event: {
+    target: { value: React.SetStateAction<string> }
+  }) => setInputEmail(event.target.value)
+
+  // Handle add button collaborator on click
+  const onAddCollaborator = () => {
+    addCollaborator.mutateAsync({
+      id: checkerId,
+      collaboratorEmail: inputEmail,
+    })
+  }
+
+  // Handle remove collaborator button on click, opens confirmation modal
+  const onRemoveCollaborator = (collaboratorEmail: string) => () => {
+    setSelectedCollaborator(collaboratorEmail)
+    onConfirmRemoveOpen()
+  }
+
+  // Handle confirm delete collaborator modal button on click
+  const onRemoveConfirm = () => {
+    deleteCollaborator.mutateAsync({
+      id: checkerId,
+      collaboratorEmail: selectedCollaborator,
+    })
+  }
+
+  // Label component with display logic for whether checker is active or not
+  const checkerActiveLabel = (hasPublished: boolean, isActive: boolean) => {
+    if (hasPublished) {
+      if (isActive) {
+        return (
+          <FormLabel htmlFor="isPublished">
+            Your checker is{' '}
+            {
+              <Text as="span" color="green">
+                active
+              </Text>
+            }{' '}
+            and accessible by the public
+            <Text color="grey" fontSize="xs">
+              Checker will show your latest published changes
+            </Text>
+          </FormLabel>
+        )
+      } else {
+        return (
+          <FormLabel htmlFor="isPublished">
+            Your checker is{' '}
+            {
+              <Text as="span" color="red">
+                inactive
+              </Text>
+            }{' '}
+            and inaccessible by the public
+            <Text color="grey" fontSize="xs">
+              Citizens will see an error message if they come across your
+              checker
+            </Text>
+          </FormLabel>
+        )
+      }
+    } else {
+      return (
+        <FormLabel htmlFor="isPublished">
+          Your checker is a draft and inaccessible by the public
+          <Text color="grey" fontSize="xs">
+            Publish your changes to activate your checker
+          </Text>
+        </FormLabel>
+      )
+    }
+  }
+
+  return (
+    <Modal isOpen onClose={onClose} size="xl">
+      <ModalOverlay />
+      <ModalContent py="16px">
```

Comment:
Maybe we can add a loading spinner while `publishedChecker` and `collaborators` are being fetched to prevent the UI from flashing between states. Can consider using `isFetched` from `react-query`.

Reply:
Added checker loading states
https://user-images.githubusercontent.com/44181699/126972467-4b3cfaa7-9aca-4036-b68b-f4c1fe0971a7.mov